include/static_math/trigonometry.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2013-2014 Morwenn
+ *
+ * static_math is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * static_math is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program. If not,
+ * see <http://www.gnu.org/licenses/>.
+ */
+
+ /*
+ * This file is the work of theLOLflashlight's GitHub account.
+ */
+
+#ifndef SMATH_TRIGONOMETRY_H_
+#define SMATH_TRIGONOMETRY_H_
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <type_traits>
+#include <static_math/cmath.h>
+#include <static_math/constants.h>
+#include <static_math/formula.h>
+
+namespace smath
+{
+    /**
+    * @brief Exponential function.
+    */
+    template<typename Float>
+    constexpr auto exp(Float x)
+        -> Float;
```

Comment:
This function is not trigonometric nor hyperbolic. It shoud be moved to the "Power and logarithmic functions" section of `static_math/cmath.h`. It can be added right before `pow`.